src/pyEQL/solution.py

```diff
@@ -1215,6 +1358,9 @@ def add_solute(self, formula: str, amount: str):
             amount (str): The amount of substance in the specified unit system. The string should contain
             both a quantity and a pint-compatible representation of a ureg. e.g. '5 mol/kg' or '0.1 g/L'.
         """
+        # Preprocess the amount string to ensure compatibility with Pint
+        amount = preprocess_units(amount)
+
```

Comment:
I think this approach could work well. In particular, if both `add_solute` and `get_amount` call `preprocess_units`, then we'll have all the logic of translating units in a single place that will ensure consistency between the two.

Reply:
See what happens if you replace this with [`interpret_units`](https://github.com/KingsburyLab/pyEQL/blob/5ef36e5f2591cb3685b5f8981bcf5038ecd28333/src/pyEQL/utils.py#L22)